File: php/src/Behavioral/Visitor/DvdInfo.php
<?

#//DvdInfo- two of three concrete Visitees

require_once 'AbstractTitleInfo.php';

class DvdInfo extends AbstractTitleInfo
{

	private $star;
	private $region;

	function __construct( $titleName, $star, $region )
	{
		parent::__construct( $titleName );
		$this->setStar( $star );
		$this->setRegion( $region);
	}

	function setStar($star) {
		$this->star = $star; 
	}
	function getStar() { return $this->star; }

	function setRegion($region) { $this->region = $region; }
	function getRegion() { return $this->region; }
   
	function accept(TitleBlurbVisitor &$tbv)
	{
		$tbv->visit($this);
	} 

}

File: csharp/src/Behavioral/Visitor/DvdInfo.cs
namespace Behavioral.Visitor {


//DvdInfo.java - two of three concrete Visitees

public class DvdInfo : AbstractTitleInfo {  
   private string star;
   private char encodingRegion;
    
   public DvdInfo(string titleName, 
                  string star, 
                  char encodingRegion) {
       this.setTitleName(titleName);
       this.setStar(star);
       this.setEncodingRegion(encodingRegion);
   }    
   
   public void setStar(string starIn) {
       this.star = starIn;
   }
   public string getStar() {
       return this.star;
   }
   public void setEncodingRegion(char encodingRegionIn) {
       this.encodingRegion = encodingRegionIn;
   }
   public char getEncodingRegion() {
       return this.encodingRegion;
   }
   
   override public void accept(TitleBlurbVisitor titleBlurbVisitor) {
       titleBlurbVisitor.visit(this);
   } 
}

}
